packages/eslint-plugin/src/rules/top-level-styles.md

```diff
@@ -0,0 +1,64 @@
+# Ensure `makeStyles`, `makeResetStyles` and `makeStaticStyles` are only called at the top-level of a file
+
+This is an opinionated rule to discourage developers from misusing the `makeStyles`, `makeResetStyles` and `makeStaticStyles` 
```

Comment:
Is this really an opinionated rule? In dev mode misusing `makeStyles` [throws an error](https://github.com/microsoft/griffel/blob/main/packages/react/src/makeStyles.ts#L14).

Reply:
I think that check just confirms it's not within a component, correct? We want to forbid it within any method at all. 

I regard it as opinionated because _technically_ these methods could be used anywhere  - as long as the dev doesn't use pass dynamic vars. But forcing it only to be used in the top-level makes it a lot less likely they do that.

Reply:
I wonder if the rule should be "no runtime parameters" in that case? Not sure how feasible that is though 😀

Reply:
I know, hey! Would've been nice to address the root cause, but that didn't seem plausible... `no-runtime-parameters` might make developers a little overconfident :)

Reply:
Text updated.